velox/exec/fuzzer/WriterFuzzer.cpp

```diff
@@ -390,47 +428,75 @@ std::string WriterFuzzer::getReferenceOutputDirectoryPath(int32_t layers) {
-    const std::string& referenceOutputDirectoryPath) {
-  const auto partitions = getPartitionNames(outputDirectoryPath);
-  LOG(INFO) << "Velox written partitions:" << std::endl;
-  for (const std::string& partition : partitions) {
-    LOG(INFO) << partition << std::endl;
+    const std::string& referenceOutputDirectoryPath,
+    int32_t bucketCount) {
+  std::vector<std::string> partitionNames;
+  const auto partitionNameAndFileCount =
+      getPartitionNameAndFilecount(outputDirectoryPath);
+  LOG(INFO) << "Velox output directory:" << outputDirectoryPath << std::endl;
+  LOG(INFO) << "Partitions and file count:" << std::endl;
+  for (const auto& i : partitionNameAndFileCount) {
+    LOG(INFO) << i.first << ":" << i.second << std::endl;
+    partitionNames.emplace_back(i.first);
   }
 
-  const auto referencedPartitions =
-      getPartitionNames(referenceOutputDirectoryPath);
-  LOG(INFO) << "Presto written partitions:" << std::endl;
-  for (const std::string& partition : referencedPartitions) {
-    LOG(INFO) << partition << std::endl;
+  std::vector<std::string> referencePartitionNames;
+  const auto referencedPartitionNameAndFileCount =
+      getPartitionNameAndFilecount(referenceOutputDirectoryPath);
+  LOG(INFO) << "Presto output directory:" << referenceOutputDirectoryPath
+            << std::endl;
+  LOG(INFO) << "Partitions and file count:" << std::endl;
+  for (const auto& i : referencedPartitionNameAndFileCount) {
+    LOG(INFO) << i.first << ":" << i.second << std::endl;
+    referencePartitionNames.emplace_back(i.first);
   }
 
-  VELOX_CHECK(
-      partitions == referencedPartitions,
-      "Velox and reference DB output directory hierarchies don't match");
+  if (bucketCount == 0) {
+    // If not bucketed, only verify if their partition names match
+    VELOX_CHECK(
+        partitionNames == referencePartitionNames,
+        "Velox and reference DB output partitions don't match");
+  } else {
+    VELOX_CHECK(
+        partitionNameAndFileCount == referencedPartitionNameAndFileCount,
+        "Velox and reference DB output partition and bucket don't match");
+  }
 }
 
-std::set<std::string> WriterFuzzer::getPartitionNames(
+// static
+std::map<std::string, int32_t> WriterFuzzer::getPartitionNameAndFilecount(
     const std::string& tableDirectoryPath) {
   auto fileSystem = filesystems::getFileSystem("/", nullptr);
   auto directories = listFolders(tableDirectoryPath);
-  std::set<std::string> partitionNames;
+  std::map<std::string, int32_t> partitionNameAndFileCount;
 
   for (std::string directory : directories) {
     // If it's a hidden directory, ignore
     if (directory.find("/.") != std::string::npos) {
       continue;
     }
 
+    // Count non-empty non-hidden files
+    auto files = fileSystem->list(directory);
+    int32_t fileCount = 0;
+    for (const auto& file : files) {
+      if (file.find("/.") == std::string::npos &&
+          fileSystem->openFileForRead(file)->size() > 0) {
```

Comment:
Why we need to check file size > 0? Is this to check if this is a directory or regular file? Please comment. Thanks!

Reply:
Commented in the code